### PR TITLE
Data Science Libraries

### LINKED ISSUE #71: Data Science Libraries Microbyte
**MicroByte Title:** Data Science Libraries

**Domain:** Data Science, Python

**Scope of learning for learners:** Hands-on knowledge of every python library needed for data science and data analytics.

**Activities:** Each Activity will contain a library, its documentation, required code for its implementation and visualization.

### PR DESCRIPTION
**Pull request by rutujak24**
## Closes #71 

<Put a x between the square brackets. Make sure you get them all>

### Checklist: 

- [x] Added activities for learners
- [x] Didn't give the solutions to the added activities straight away
- [x] Followed a LearnByDoing approach
- [x] Clean and well documented code (if source code files are involved)
- [x] Added images for better understanding if deemed necessary

### CODE REVIEW

**Comment by YourAKShaw** [comment]
@rutujak24 I appreciate your generous efforts, but I need to break it to you that like every Organization and repo out there, we've got some contribution guidelines which you need to follow. Kindly go through the [**CONTRIBUTING.md**](https://github.com/Crio-Bytes/Miscellaneous/blob/main/CONTRIBUTING.md) of this repo.

For a quick go through, you need to make a MicroByte proposal in a manner similar to https://github.com/Crio-Bytes/Miscellaneous/issues/3. Kindly utilize the `Add MicroByte` issue template for this.

**You don't need to wait for us to assign it to you**. 

Then you need to make a PR in a manner similar to https://github.com/Crio-Bytes/Miscellaneous/pull/1. Kindly utilize [this](https://github.com/Crio-Bytes/Miscellaneous/blob/main/.github/PULL_REQUEST_TEMPLATE/adding-microbyte.md) PR template for the same.

Basically that's it. For more clearance, you can checkout this [YT video](https://www.youtube.com/watch?v=qdSn8v9nU0g&feature=youtu.be).

Also, you're free to reach out to us when you need. :wave:

**Comment by rutujak24** [comment]
> @rutujak24 I appreciate your generous efforts, but I need to break it to you that like every Organization and repo out there, we've got some contribution guidelines which you need to follow. Kindly go through the [**CONTRIBUTING.md**](https://github.com/Crio-Bytes/Miscellaneous/blob/main/CONTRIBUTING.md) of this repo.
> 
> For a quick go through, you need to make a MicroByte proposal in a manner similar to #3. Kindly utilize the `Add MicroByte` issue template for this.
> 
> **You don't need to wait for us to assign it to you**.
> 
> Then you need to make a PR in a manner similar to #1. Kindly utilize [this](https://github.com/Crio-Bytes/Miscellaneous/blob/main/.github/PULL_REQUEST_TEMPLATE/adding-microbyte.md) PR template for the same.
> 
> Basically that's it. For more clearance, you can checkout this [YT video](https://www.youtube.com/watch?v=qdSn8v9nU0g&feature=youtu.be).
> 
> Also, you're free to reach out to us when you need. 👋

All Done!

**Review by YourAKShaw — CHANGES_REQUESTED** [review]
@rutujak24 **appreciate your efforts**, though we'll need to work out some changes.

- The **README.md** is kinda messy with explanations that could've been **easily Googled**.
- The **README.md** contains no mention of the activities and briefing of the same.

So, **what needs to be done then?**

- The **README.md** needs a complete makeover. :sweat_smile: When I say makeover, kindly don't add details which could've been easily searched on the Web. You can add a few points, but really few points.
- Regarding what I mentioned earlier about the activities, in the **README.md** kindly add enough info about the activities that can get any learner scratching their heads and they'll work around the solution themselves, provided that the prerequisites are already met by the learner, which you can mention in the **README.md** itself.
- Follow a **LearnByDoing** approach, where the learner shouldn't be given the solution right away, but enough info should be provided that a learner can make out the solution on her own.

Moreover, highly appreciate your work until this point :100: :100: , just more fine tuning to your content and we'll be happy to merge it!

If you need further clarification, feel free to reach out to us on slack!

Cheers. :clinking_glasses:

**Comment by amoghrajesh** [comment]
> @rutujak24 **appreciate your efforts**, though we'll need to work out some changes.
> 
> * The **README.md** is kinda messy with explanations that could've been **easily Googled**.
> * The **README.md** contains no mention of the activities and briefing of the same.
> 
> So, **what needs to be done then?**
> 
> * The **README.md** needs a complete makeover. 😅 When I say makeover, kindly don't add details which could've been easily searched on the Web. You can add a few points, but really few points.
> * Regarding what I mentioned earlier about the activities, in the **README.md** kindly add enough info about the activities that can get any learner scratching their heads and they'll work around the solution themselves, provided that the prerequisites are already met by the learner, which you can mention in the **README.md** itself.
> * Follow a **LearnByDoing** approach, where the learner shouldn't be given the solution right away, but enough info should be provided that a learner can make out the solution on her own.
> 
> Moreover, highly appreciate your work until this point 💯 💯 , just more fine tuning to your content and we'll be happy to merge it!
> 
> If you need further clarification, feel free to reach out to us on slack!
> 
> Cheers. 🥂

Okay @Ak-Shaw has covered all I wanted to say @rutujak24. Great work though. 
Make these changes and we shall re review it ASAP 😄

**Comment by rutujak24** [comment]
Hey, @Ak-Shaw @amoghrajesh I have done all the required changes and added high-quality content and code that I generally use in my real-tie projects and hackathons. All the suggested changes are also done. I guess it is good to be merged now :)

**Comment by YourAKShaw** [comment]
> @rutujak24 **appreciate your efforts**, though we'll need to work out some changes.
> 
>     * The **README.md** is kinda messy with explanations that could've been **easily Googled**.
> 
>     * The **README.md** contains no mention of the activities and briefing of the same.
> 
> 
> So, **what needs to be done then?**
> 
>     * The **README.md** needs a complete makeover. sweat_smile When I say makeover, 
>        kindly don't add details which could've been easily searched on the Web. You can add a 
>        few points, but really few points.
> 
>     * Regarding what I mentioned earlier about the activities, in the **README.md** kindly add 
>       enough info about the activities that can get any learner scratching their heads and they'll work 
>       around the solution themselves, provided that the prerequisites are already met by the learner, 
>       which you can mention in the **README.md** itself.
> 
>     * Follow a **LearnByDoing** approach, where the learner shouldn't be given the solution right 
>       away, but enough info should be provided that a learner can make out the solution on her own.
> 
> 
> Moreover, highly appreciate your work until this point 100 100 , just more fine tuning to your content and we'll be happy to merge it!
> 
> If you need further clarification, feel free to reach out to us on slack!
> 
> Cheers. clinking_glasses

Following what I requested earlier, I don't see any significant changes you've made. Kindly don't get me wrong, but your **README.md** seems like data dumping, where as we need a content which engages the learners. Kindly go through this entire message carefully and if you still don't understand then you are free to ping me up on slack.

**Comment by rutujak24** [comment]
> > @rutujak24 **appreciate your efforts**, though we'll need to work out some changes.
> > ```
> > * The **README.md** is kinda messy with explanations that could've been **easily Googled**.
> > 
> > * The **README.md** contains no mention of the activities and briefing of the same.
> > ```
> > 
> > 
> > So, **what needs to be done then?**
> > ```
> > * The **README.md** needs a complete makeover. sweat_smile When I say makeover, 
> >    kindly don't add details which could've been easily searched on the Web. You can add a 
> >    few points, but really few points.
> > 
> > * Regarding what I mentioned earlier about the activities, in the **README.md** kindly add 
> >   enough info about the activities that can get any learner scratching their heads and they'll work 
> >   around the solution themselves, provided that the prerequisites are already met by the learner, 
> >   which you can mention in the **README.md** itself.
> > 
> > * Follow a **LearnByDoing** approach, where the learner shouldn't be given the solution right 
> >   away, but enough info should be provided that a learner can make out the solution on her own.
> > ```
> > 
> > 
> > Moreover, highly appreciate your work until this point 100 100 , just more fine tuning to your content and we'll be happy to merge it!
> > If you need further clarification, feel free to reach out to us on slack!
> > Cheers. clinking_glasses
> 
> Following what I requested earlier, I don't see any significant changes you've made. Kindly don't get me wrong, but your **README.md** seems like data dumping, where as we need a content which engages the learners. Kindly go through this entire message carefully and if you still don't understand then you are free to ping me up on slack.

The is the final PR is #73 .
All the requested changes are done in this.
Please check

**Comment by YourAKShaw** [comment]
@rutujak24 Kindly close one of the two PRs. The choice is yours. Then we can guide you based on our reviewing.
Also, always make changes in one PR, as it makes it much more easier to manage.

**Comment by rutujak24** [comment]
Hey @Ak-Shaw Please close this PR and review #73
